docs/architecture/adr-050-sign-mode-textual.md

```diff
@@ -181,25 +229,9 @@ When this option is set on a `Msg`, a registered function will transform the `Ms
 
 The `<unique algorithm identifier>` is a string convention chosen by the application developer and is used to identify the custom `Msg` renderer. For example, the documentation or specification of this custom algorithm can reference this identifier. This identifier CAN have a versioned suffix (e.g. `_v1`) to adapt for future changes (which would be consensus-breaking). We also recommend adding Protobuf comments to describe in human language the custom logic used.
 
-Moreover, the renderer must provide 2 functions: one for formatting from Protobuf to string, and one for parsing string to Protobuf. These 2 functions are provided by the application developer. To satisfy point #1, these 2 functions MUST be bijective with each other. Bijectivity of these 2 functions will not be checked by the SDK at runtime. However, we strongly recommend the application developer to include a comprehensive suite in their app repo to test bijectivity, as to not introduce security bugs. A simple bijectivity test looks like:
-
-```
-// for renderer, msg, and ctx of the right type
-keyvals, err := renderer.Format(ctx, msg)
-if err != nil {
-    fail_check()
-}
-msg2, err := renderer.Parse(ctx, keyvals)
-if err != nil {
-    fail_check()
-}
-if !proto.Equal(msg, msg2) {
-    fail_check()
-}
-pass_check()
-```
+Moreover, the renderer must provide 2 functions: one for formatting from Protobuf to string, and one for parsing string to Protobuf. These 2 functions are provided by the application developer. To satisfy point #1, these 2 functions MUST be bijective with each other. Bijectivity of these 2 functions will not be checked by the SDK at runtime. However, we strongly recommend the application developer to include a comprehensive suite in their app repo to test bijectivity, as to not introduce security bugs.
```

Comment:
If the application fails to ensure bijectivity, are any guarantees asserted by Cosmos and/or Tendermint violated?